components/AddContentProviderForm/AddContentProviderForm.tsx

```diff
@@ -0,0 +1,64 @@
+import type { ChangeEventHandler, FormEventHandler } from 'react';
+import { useCallback, useState } from 'react';
+
+import { Button } from '../Button/Button';
+
+import styles from './addContentProviderForm.module.scss';
+import { useFormReducer } from './utils';
+
+export const AddContentProviderForm = () => {
+  const [contentURL, setContentURL] = useState('');
```

Comment:
Nie lepiej trzymać wszystkie pola w jednym stanie, a następnie w łatwiejszy sposób mieć do nich dostęp? Mam tu na myśli:

```ts
const [fields, setFields] = useState({ contentURL: '', email: '' })
```

Reply:
To jest trochę staromodne podejście z czasów, gdy jeszcze używało się komponentów klasowych, React rekomenduje używanie osobnych `state` w hook'ach.

Reply:
IMO w przypadku takiego forma będzie to wyglądać lepiej niż powtarzanie handlerów. Od strony wydajności - brak różnicy, i tak komponent jest ponownie renderowany